hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AutoAccountCreator.java

```diff
@@ -0,0 +1,239 @@
+     */
+    public void create(@NonNull final Bytes alias, final boolean isByTokenTransfer) {
+        final var accountsConfig = handleContext.configuration().getConfigData(AccountsConfig.class);
+
+        validateTrue(
+                accountStore.sizeOfAccountState() + 1 <= accountsConfig.maxNumber(),
+                ResponseCodeEnum.MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED);
+
+        final var config = handleContext.configuration().getConfigData(AutoCreationConfig.class);
+        validateTrue(config.enabled(), NOT_SUPPORTED);
+
+        final var tokensConfig = handleContext.configuration().getConfigData(TokensConfig.class);
+        if (isByTokenTransfer) {
+            validateTrue(tokensConfig.autoCreationsIsEnabled(), ResponseCodeEnum.NOT_SUPPORTED);
+        }
+
+        final TransactionBody.Builder syntheticCreation;
+        String memo;
+
+        // checks tokenAliasMap if the change consists an alias that is already used in previous
+        // iteration of the token transfer list. This map is used to count number of
+        // maxAutoAssociations needed on auto created account
+        if (isByTokenTransfer) {
+            tokenAliasMap.putIfAbsent(alias, Collections.emptySet());
+        }
+
+        final var maxAutoAssociations =
+                tokenAliasMap.getOrDefault(alias, Collections.emptySet()).size();
+        final var isAliasEVMAddress = EntityIdUtils.isOfEvmAddressSize(alias);
+        if (isAliasEVMAddress) {
+            syntheticCreation = createHollowAccount(alias, 0L);
+            memo = LAZY_MEMO;
+        } else {
+            final var key = asKeyFromAlias(alias);
+            syntheticCreation = createAccount(alias, key, 0L, maxAutoAssociations);
+            memo = AUTO_MEMO;
+        }
+
+        var fee = autoCreationFeeFor(syntheticCreation);
+        if (isAliasEVMAddress) {
+            fee += getLazyCreationFinalizationFee();
+        }
+        // TODO: Check if payer has enough balance to pay for the fee
+
+        final var childRecord = handleContext.dispatchRemovableChildTransaction(
+                syntheticCreation.memo(memo).build(), CryptoCreateRecordBuilder.class);
+
+        if (!isAliasEVMAddress) {
+            final var key = asKeyFromAlias(alias);
+            if (key.hasEcdsaSecp256k1()) {
+                final var evmAddress = tryAddressRecovery(key, EthSigsUtils::recoverAddressFromPubKey);
+                if (evmAddress != null) {
+                    childRecord.evmAddress(Bytes.wrap(evmAddress));
+                }
+            }
+        }
+        // TODO: Not sure if fee should be set here childRecord.transactionFee(fee);
+    }
+
+    /**
+     * Get fees for finalization of lazy creation.
+     * @return fee for finalization of lazy creation
+     */
+    private long getLazyCreationFinalizationFee() {
+        final var updateTxnBody = CryptoUpdateTransactionBody.newBuilder()
+                .key(Key.newBuilder().ecdsaSecp256k1(Bytes.EMPTY).build());
```

Comment:
I'd define this `CryptoTransactionBody` as a static constant so we don't have to recreate it every time

Reply:
Done